src/Microsoft.Diagnostics.Monitoring/ClientEndpointInfoSource.cs

```diff
@@ -15,12 +16,22 @@ internal sealed class ClientEndpointInfoSource : IEndpointInfoSourceInternal
     {
         public Task<IEnumerable<IEndpointInfo>> GetEndpointInfoAsync(CancellationToken token)
         {
+            bool isInDockerContainer = RuntimeInfo.IsInDockerContainer;
+            int currentPid = Process.GetCurrentProcess().Id;
+
             List<IEndpointInfo> endpointInfos = new List<IEndpointInfo>();
             foreach (int pid in DiagnosticsClient.GetPublishedProcesses())
             {
-                // CONSIDER: Generate a "runtime instance identifier" based on the pipe name
-                // e.g. pid + disambiguator in GUID form.
-                endpointInfos.Add(new EndpointInfo(pid));
+                // If in Docker container, allow adding all processes to the list because
+                // multi-container environments could have the same process ID. Expect that
+                // user to have set COMPLUS_EnableDiagnostics=0 for this process. Otherwise,
```

Comment:
I'm leary of this expectation to disable diagnostics, added my thoughts to the discussion below.

Reply:
We still need to rely on that expectation if using the mounted /tmp strategy in a multi-container scenario, but otherwise it is not necessary with the new filter heuristic I've added.